[mono] Allow the use of an override directory for LLVM.

imhameed opened the pull request:
With this change, a custom LLVM directory can be set via the `MonoLLVMDir` property. When using a custom LLVM directory, llvm-init.proj will be ignored.

`--enable--llvm` (with two dashes) is not a valid mono configure argument, and `--enable-llvm` is implied by `--with-llvm`.

imhameed committed:
Allow the use of an override directory for LLVM.

With this change, when `MonoLLVMDir` is set and `MonoEnableLLVM` is not
true, `MonoLLVMDir` will be passed directly to the mono configure script
and llvm-init.proj will be ignored.

`--enable--llvm` (with two dashes) is not a valid mono configure
argument, and `--enable-llvm` is implied by `--with-llvm`.

akoeplinger commented on src/mono/mono.proj:
```diff
@@ -33,7 +33,7 @@
     </PropertyGroup>
     <PropertyGroup>
-        <_MonoConfigureParams Condition="'$(MonoEnableLLVM)' == 'true'">$(_MonoConfigureParams) --enable--llvm --with-llvm=$(MonoLLVMDir) </_MonoConfigureParams>
+        <_MonoConfigureParams Condition="'$(MonoLLVMDir)' != ''">$(_MonoConfigureParams) --with-llvm=$(MonoLLVMDir) </_MonoConfigureParams>
```
the more canonical way to do this would be:

```diff
-<MonoLLVMDir Condition="'$(MonoEnableLLVM)' == 'true'">$(MonoObjDir)llvm</MonoLLVMDir>
+<MonoLLVMDir Condition="'$(MonoEnableLLVM)' == 'true' and '$(MonoLLVMDir)' == ''">$(MonoObjDir)llvm</MonoLLVMDir>
```

akoeplinger replied:
ah sorry, scratch that, I missed the part of your PR description where you say `--enable--llvm` is invalid :)

imhameed replied:
Done.

akoeplinger reviewed (changes requested):
we should still require `/p:MonoEnableLLVM=true` but allow you to override the `MonoLLVMDir`.

imhameed committed:
Require the use of `MonoEnableLLVM` for any LLVM-enabled mono build.

Skip llvm-init.proj if `MonoLLVMDir` is not the same as the default
directory used by `MonoEnableLLVM`.
